apps/api/src/app/widgets/widgets.controller.ts

```diff
@@ -102,36 +99,75 @@ export class WidgetsController {
     @Query('feedIdentifier') feedId: string[] | string,
     @Query('seen') seen: boolean
   ): Promise<UnseenCountResponse> {
-    let feedsQuery: string[];
-    if (feedId) {
-      feedsQuery = Array.isArray(feedId) ? feedId : [feedId];
-    }
+    const feedsQuery = this.toArray(feedId);
 
-    const command = GetUnseenCountCommand.create({
+    const command = GetFeedCountCommand.create({
       organizationId: subscriberSession._organizationId,
       subscriberId: subscriberSession.subscriberId,
       environmentId: subscriberSession._environmentId,
       feedId: feedsQuery,
       seen,
     });
 
-    return await this.genUnseenCountUsecase.execute(command);
+    return await this.getFeedCountUsecase.execute(command);
+  }
+
+  @UseGuards(AuthGuard('subscriberJwt'))
+  @Get('/notifications/count')
+  async getCount(
+    @SubscriberSession() subscriberSession: SubscriberEntity,
+    @Query() query: GetCountQuery
+  ): Promise<UnseenCountResponse> {
+    const feedsQuery = this.toArray(query.feedIdentifier);
+
+    const command = GetFeedCountCommand.create({
+      organizationId: subscriberSession._organizationId,
+      subscriberId: subscriberSession.subscriberId,
+      environmentId: subscriberSession._environmentId,
+      feedId: feedsQuery,
+      seen: query.seen,
+      read: query.read,
+    });
+
+    return await this.getFeedCountUsecase.execute(command);
   }
 
   @UseGuards(AuthGuard('subscriberJwt'))
   @Post('/messages/:messageId/seen')
```

Comment:
Duplicating this endpoint and marking this as deprecated on swagger. The new endpoint will get messagedIds in the body of the request.